core/src/main/java/com/datastax/oss/driver/internal/core/session/DefaultSession.java

```diff
@@ -107,11 +106,50 @@ private DefaultSession(
     this.metricUpdater = context.metricUpdaterFactory().newSessionUpdater();
   }
 
-  private CompletionStage<CqlSession> init(CqlIdentifier keyspace) {
+  public CompletionStage<CqlSession> init(CqlIdentifier keyspace) {
+    DriverConfigProfile defaultConfig = context.config().getDefaultProfile();
+    if (defaultConfig.isDefined(DefaultDriverOption.STARTUP_CUSTOM_BANNER)) {
+      printCustomBanner(defaultConfig.getString(DefaultDriverOption.STARTUP_CUSTOM_BANNER));
+    }
+    if (defaultConfig.getBoolean(DefaultDriverOption.STARTUP_PRINT_BASIC_INFO)) {
+      printBasicStartupInfo();
+    }
     RunOrSchedule.on(adminExecutor, () -> singleThreaded.init(keyspace));
     return singleThreaded.initFuture;
   }
 
+  protected void printCustomBanner(String banner) {
+    STARTUP_LOG.info(banner);
+  }
+
+  protected void printBasicStartupInfo() {
+    try {
+      DriverInfo driverInfo = getDriverInfo();
+      int processId = -1;
+      if (Native.isGetProcessIdAvailable()) {
+        processId = Native.getProcessId();
+      }
+      long start = System.nanoTime();
+      STARTUP_LOG.info("[{}] {} initializing (PID: {})", logPrefix, driverInfo, processId);
+      singleThreaded.initFuture.thenRun(
+          () -> {
+            Duration elapsed =
+                Duration.ofMillis(TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start));
+            String elapsedStr =
+                elapsed.toString().substring(2).replaceAll("(\\d[HMS])(?!$)", "$1 ").toLowerCase();
+            int size = getPools().size();
+            STARTUP_LOG.info(
+                "[{}] Session successfully initialized in {}, connected to {} node{}",
+                logPrefix,
+                elapsedStr,
+                size,
+                size > 1 ? "s" : "");
```

Comment:
I would prefer to have a single message (so the second one after the session is initialized).

Initialization time sounds nice to have, not sure if the PID will be that useful but why not. If we go with my suggestion on the single configuration option, these would have to be added to the list of custom interpolation variables.